sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.ts

```diff
@@ -432,4 +422,35 @@ export class StudentAESTController extends BaseController {
       userToken.userId,
     );
   }
+
+  /**
+   * Updates the student info (lastname, givenNames, dob, email)
+   * for a basic BCeID student.
+   * @param studentId related student id.
+   * @param payload payload to be updated.
+   */
+  @Roles(Role.StudentEditProfile)
+  @Patch(":studentId")
+  @ApiNotFoundResponse({ description: "Student does not exist." })
+  async updateProfileInformation(
+    @Param("studentId", ParseIntPipe) studentId: number,
+    @Body() payload: UpdateStudentDetailsAPIInDTO,
+    @UserToken() userToken: IUserToken,
+  ): Promise<void> {
+    const student = await this.studentService.getStudentById(studentId);
```

Comment:
Please see the recommendation below regarding the execution order.
In the below, the validations will happen at the beginning of the method and the core update will happen at the last part if everything is valid.
Notice that the main method will not even be enclosed in an `if` condition.
```ts
const student = await this.studentService.getStudentById(studentId);
if (!student) {
  throw new NotFoundException("Student does not exist.");
}
if (student.user.identityProviderType !== IdentityProviders.BCeIDBasic) {
  throw new UnprocessableEntityException(
    "Not possible to update a non-basic-BCeID student.",
  );
}
await this.studentService.updateStudentUserData({
  studentId,
  userId: userToken.userId,
  ...payload,
});
```

Reply:
Do we want a specific message like `"Not possible to update a non-basic-BCeID student.",` or a general message like `Student does not exist.` will work in case of a non basic BCeID student?